scheduler/src/cook/kubernetes/compute_cluster.clj

```diff
@@ -290,6 +293,8 @@
         ^V1Pod pod (api/task-metadata->pod pod-namespace compute-cluster task-metadata)
         new-cook-expected-state-dict {:cook-expected-state :cook-expected-state/starting
                                       :launch-pod {:pod pod}}]
+    (when (api/synthetic-pod? pod-name)
+      (cache/put-cache! caches/instance-uuid->job-uuid identity pod-name (get-in task-metadata [:task-request :job :job/uuid])))
```

Comment:
I think it'd be nice to include a comment explaining why this is needed / giving some context for it.

Reply:
Do you mean specifically the `(when (api/synthetic-pod? pod-name)` part? Or do you just mean to provide context on why the cache exists.

Reply:
I meant context on why the cache exists (but maybe the synthetic pod check warrants a little explanation too)